gitpod-io/gitpod
# Update the `sub` claim

## Description

Adds the organization and user ID along with the context URL to the subject claim of our OIDC JWT token.

## Related Issue(s)

Fixes IDE-103

## How to test

In the preview environment, create a workspace and execute `gp idp token --audience gitpod.io` within it. After that, inspect the outputted token with a tool like https://jwt.io to verify it has all the fields this PR promises.

If you're not into testing yourself, I did it for you already. Here's the decoded token:
```json
{
  "iss": "https://api.ft-sub-update.preview.gitpod-dev.com/idp",
  "aud": [
    "gitpod.io"
  ],
  "azp": "gitpod.io",
  "c_hash": "MbIWgwnQBOAorEi9dN1-UQ",
  "exp": 1687627102,
  "iat": 1687623502,
  "auth_time": 1687623502,
  "email": "filip@gitpod.io",
  "name": "filiptronicek",
  "org_id": "6730b752-6dd6-48d0-9bc1-e8ac168d0f13",
  "sub": "org_id:6730b752-6dd6-48d0-9bc1-e8ac168d0f13:user_id:9a12464b-eb40-4879-8437-710fcb585441:context:https%3A%2F%2Fgithub.com%2Fcsweichel%2Foci-tool"
}
```

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-sub-update</li>
	<li><b>🔗 URL</b> - <a href="https://ft-sub-update.preview.gitpod-dev.com/workspaces" target="_blank">ft-sub-update.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold
